icu4c/source/common/locid.cpp

```diff
@@ -717,7 +759,7 @@ Locale::createFromName (const char *name)
     if (name) {
         Locale l("");
         l.init(name, FALSE);
-        return l;
+        return static_cast<Locale&&>(l);
```

Comment:
When returning by value, you get Return Value Optimization.  The Locale is constructed in-place in the caller's stack.  That's even more efficient than an std::move.

https://shaharmike.com/cpp/rvo/

Reply:
Oh, I hadn't realized that RVO happened here, where there are two different possible return objects.